dns/jsonify/main.go

```diff
@@ -0,0 +1,218 @@
+		if err != nil {
+			return err
+		}
+		latency.DataItems = appendLatency(latency.DataItems, result)
+		latencyPerc.DataItems = appendLatencyPerc(latencyPerc.DataItems, result)
+		queries.DataItems = appendQueries(queries.DataItems, result)
+	}
+
+	timeString := time.Now().Format(time.RFC3339)
+	if err = saveMetric(&latency, filepath.Join(jsonDirPath, "Latency_"+benchmarkName+"_"+timeString+".json")); err != nil {
+		return err
+	}
+	if err = saveMetric(&latencyPerc, filepath.Join(jsonDirPath, "LatencyPerc_"+benchmarkName+"_"+timeString+".json")); err != nil {
+		return err
+	}
+	if err = saveMetric(&queries, filepath.Join(jsonDirPath, "Queries_"+benchmarkName+"_"+timeString+".json")); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getFileList(dir string) ([]string, error) {
+	var fileNames []string
+	files, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return fileNames, err
+	}
+
+	for _, file := range files {
+		if !file.IsDir() && filepath.Ext(file.Name()) == ".out" {
+			fileNames = append(fileNames, file.Name())
+		}
+	}
+	return fileNames, nil
+}
+
+func readBenchamrkResult(path string) (*BenchamrkResult, error) {
+	var result BenchamrkResult
+	bin, err := ioutil.ReadFile(path)
+	if err != nil {
+		return nil, fmt.Errorf("reading error: %v", err)
+	}
+
+	if err := yaml.NewYAMLOrJSONDecoder(bytes.NewBuffer(bin), 4096).Decode(&result); err != nil {
+		return nil, fmt.Errorf("decoding failed: %v", err)
+	}
+	return &result, nil
+}
+
+func appendLatency(items []perftype.DataItem, result *BenchamrkResult) []perftype.DataItem {
+	return append(items, perftype.DataItem{
+		Unit: "ms",
+		Labels: map[string]string{
+			"max_qps":    getString(result.Params, "max_qps"),
+			"query_file": getString(result.Params, "query_file"),
+		},
+		Data: map[string]float64{
+			"max_latency": getFloat(result.Data, "max_latency") * S_TO_MS,
+			"avg_latency": getFloat(result.Data, "avg_latency") * S_TO_MS,
+			"min_latency": getFloat(result.Data, "min_latency") * S_TO_MS,
+		},
+	})
+}
+
+func appendLatencyPerc(items []perftype.DataItem, result *BenchamrkResult) []perftype.DataItem {
+	return append(items, perftype.DataItem{
+		Unit: "ms",
+		Labels: map[string]string{
+			"max_qps":    getString(result.Params, "max_qps"),
+			"query_file": getString(result.Params, "query_file"),
+		},
+		Data: map[string]float64{
+			"Perc50": getFloat(result.Data, "latency_50_percentile"),
+			"Perc90": getFloat(result.Data, "latency_95_percentile"),
+			"Perc99": getFloat(result.Data, "latency_99_percentile"),
+		},
+	})
+}
+
+func appendQueries(items []perftype.DataItem, result *BenchamrkResult) []perftype.DataItem {
+	return append(items, perftype.DataItem{
+		Unit: "",
+		Labels: map[string]string{
+			"max_qps":    getString(result.Params, "max_qps"),
+			"query_file": getString(result.Params, "query_file"),
+		},
+		Data: map[string]float64{
+			"queries_completed": getFloat(result.Data, "queries_completed"),
```

Comment:
Create a golang struct with the right fields for deserializing into instead of operating on interface{} types

Reply:
Done.
I've also had to do slight changes in parsing method.